checks/frozen_deps_test.go

```diff
@@ -90,12 +92,14 @@ func TestGithubWorkflowPinning(t *testing.T) {
 					panic(fmt.Errorf("cannot read file: %w", err))
 				}
 			}
-			r, err := validateGitHubActionWorkflow(tt.args.Filename, content, tt.args.Logf)
+			r, err := validateGitHubActionWorkflow(tt.args.Filename, content, tt.args.Log.Logf)
 
 			if !errors.Is(err, tt.want.Error) ||
-				r != tt.want.Result {
-				t.Errorf("TestGithubWorkflowPinning:\"%v\": %v (%v,%v) want (%v, %v)",
-					tt.name, tt.args.Filename, r, err, tt.want.Result, tt.want.Error)
+				r != tt.want.Result ||
+				len(tt.args.Log.messages) != tt.want.NumberOfErrors {
+				t.Errorf("TestDockerfileScriptDownload:\"%v\": %v (%v,%v,%v) want (%v, %v, %v)\n%v",
+					tt.name, tt.args.Filename, r, err, len(tt.args.Log.messages), tt.want.Result, tt.want.Error, tt.want.NumberOfErrors,
```

Comment:
nit: Do we need that complicated message, similar is better ?

Reply:
this gives all information we need to fix the problem. It only triggers when there's a failure.